### PR TITLE
Flag Fixes

### LINKED ISSUE #14128: Flag Bugs
you can rip down flags at a distance, i.e. click flag, get menu, walk away, rip down flag somehow.
you don't get the flag back if you rip it down.

### PR DESCRIPTION
**Pull request by Doxxmedearly**
Fixes #14128

Flags were operating on sign code, which is old and stupid.
Changes:
-Makes a proper icon when the flag is town down by adding an alpha mask. The current ripped icon is totally broken.
-Makes it so you can't tear down flags from a distance
-Made flags actually give you the flag item back, instead of some weird quasi-sign item.
-Made large flags link up and reference each other so that changes to one half can be applied to the other. 
-General quality updates, like visible messages, sounds, code organization, etc.

### CODE REVIEW

**Comment by Doxxmedearly** [comment]
!review

**Comment by Doxxmedearly** [comment]
!bugfix

File: code/game/objects/structures/signs.dm [hunk 1]
```diff
@@ -7,26 +7,16 @@
...
-		else
-	return
+	qdel(src)
 
-/obj/structure/sign/attackby(obj/item/tool as obj, mob/user as mob)	//deconstruction
+/obj/structure/sign/attackby(obj/item/tool, mob/user)	//deconstruction
 	if(tool.isscrewdriver() && !istype(src, /obj/structure/sign/double))
-		to_chat(user, "You unfasten the sign with your [tool].")
-		unfasten()
+		playsound(get_turf(user), tool.usesound, 50, 1)
```
Comment (alsoandanswer) [inline code/game/objects/structures/signs.dm:14]: I suggest using the new use_tool proc here.

Comment (Doxxmedearly) [reply code/game/objects/structures/signs.dm:14]: Every implementation of use_tool I've seen is for tasks that involve a `delay` time (And indeed, the proc returns immediately without a `delay` set). This action not have a delay time.

**Review by alsoandanswer — APPROVED** [review]

**Pull request merged**